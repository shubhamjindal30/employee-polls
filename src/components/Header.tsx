import { useHistory, useLocation } from 'react-router-dom';
import { AppBar, Box, Button, Typography, makeStyles } from '@material-ui/core';

const routes = [
  {
    route: '/home',
    title: 'Home'
  },
  {
    route: '/leaderboard',
    title: 'Leaderboard'
  },
  {
    route: '/new',
    title: 'New'
  }
];

const Header = () => {
  const classes = useStyles();
  const history = useHistory();
  const location = useLocation();

  return (
    <AppBar className={classes.appbar}>
      <Box>
        {routes.map((route) => (
          <Button
            key={route.route}
            className={classes.button}
            style={{ borderBottom: location.pathname === route.route ? '2px solid #fff' : '2px solid transparent' }}
            onClick={() => history.push(route.route)}
          >
            {route.title}
          </Button>
        ))}
      </Box>
      <Box style={{ display: 'flex', flexDirection: 'row', alignItems: 'center' }}>
        <Typography>User 1</Typography>
        <Button className={classes.button}>Logout</Button>
      </Box>
    </AppBar>
  );
};

export default Header;

const useStyles = makeStyles(() => ({
  appbar: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between'
  },
  button: {
    color: '#fff',
    textTransform: 'none',
    borderRadius: 0
  }
}));
