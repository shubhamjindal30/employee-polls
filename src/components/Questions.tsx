import { Grid, Typography, makeStyles, Divider, Button } from '@material-ui/core';
import { useHistory } from 'react-router-dom';
import moment from 'moment';

import { Question } from '../store/questions/types';

interface QuestionsProps {
  title: string;
  questions: Question[];
}

const Questions: React.FunctionComponent<QuestionsProps> = ({ title, questions }) => {
  const classes = useStyles();
  const history = useHistory();

  return (
    <Grid className={classes.container} container direction="column">
      <Grid className={classes.headingContainer} item>
        <Typography variant="h6">{title}</Typography>
      </Grid>
      <Grid
        className={classes.questionContainer}
        item
        container
        justifyContent="space-evenly"
        alignItems="stretch"
      >
        {questions.map((question) => (
          <Grid key={question.id} className={classes.questionCard} xs={3}>
            <Typography className={classes.authorText} variant="body1">
              {question.author}
            </Typography>
            <Typography variant="caption" color="textSecondary">
              {moment(question.timestamp).format('h:mm A | DD/MM/YYYY')}
            </Typography>
            <Divider className={classes.divider} />
            <Button
              variant="contained"
              color="primary"
              fullWidth
              onClick={() => history.push(`/questions/${question.id}`)}
            >
              Show
            </Button>
          </Grid>
        ))}
      </Grid>
    </Grid>
  );
};

export default Questions;

const useStyles = makeStyles({
  container: {
    border: '1px solid lightgray',
    borderRadius: 5,
    marginBottom: 20
  },
  headingContainer: {
    padding: 10,
    textAlign: 'center',
    borderBottom: '1px solid lightgray'
  },
  questionContainer: {
    padding: 15
  },
  questionCard: {
    marginTop: 10,
    marginBottom: 10,
    padding: 10,
    paddingTop: 15,
    textAlign: 'center',
    borderRadius: 5,
    border: '1px solid lightgray'
  },
  authorText: {
    fontWeight: 'bold'
  },
  divider: {
    marginTop: 15,
    marginBottom: 10
  }
});
