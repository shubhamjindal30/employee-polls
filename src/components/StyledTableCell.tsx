import { TableCell, TableCellProps, makeStyles } from '@material-ui/core';

const StyledTableCell = (props: TableCellProps) => {
  const classes = useStyles();
  return (
    <TableCell className={classes.styledCell} {...props}>
      {props.children}
    </TableCell>
  );
};

const useStyles = makeStyles((theme) => ({
  styledCell: {
    backgroundColor: theme.palette.primary.dark,
    color: theme.palette.primary.contrastText,
    fontWeight: 'bold',
    fontSize: 'medium'
  }
}));

export default StyledTableCell;
