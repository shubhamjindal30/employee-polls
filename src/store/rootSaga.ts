import { all, fork } from 'redux-saga/effects';

import { watchUserRequests } from './users/saga';
import { watchAuthRequests } from './auth/saga';

export function* watcherSaga() {
  yield all([fork(watchUserRequests), fork(watchAuthRequests)]);
}
