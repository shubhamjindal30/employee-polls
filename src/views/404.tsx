import { withRouter } from 'react-router-dom';
import { Container, makeStyles, Typography } from '@material-ui/core';

const PageNotFound = () => {
  const classes = useStyles();

  return <Container className={classes.container}>
    <Typography variant='h5'>Page not found!</Typography>
  </Container>;
};

export default withRouter(PageNotFound);

const useStyles = makeStyles({
  container: {
    marginTop: 60
  }
});
