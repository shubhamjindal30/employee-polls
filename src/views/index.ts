import Login from './Login';
import Home from './Home';
import PageNotFound from './404';
import NewQuestion from './NewQuestion';

export { Login, Home, PageNotFound, NewQuestion };
