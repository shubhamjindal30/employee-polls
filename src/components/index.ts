import Header from './Header';
import Questions from './Questions';

export { Header, Questions };
