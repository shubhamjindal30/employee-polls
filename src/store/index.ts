import { createStore, applyMiddleware, combineReducers } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';

import { watcherSaga } from './rootSaga';
import userReducer from './users/reducer';
import authReducer from './auth/reducer';

const rootReducer = combineReducers({
  user: userReducer,
  auth: authReducer
});

const sagaMiddleware = createSagaMiddleware();

const middlewares = [sagaMiddleware];

const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(...middlewares)));

sagaMiddleware.run(watcherSaga);

export type RootState = ReturnType<typeof rootReducer>;

export default store;
