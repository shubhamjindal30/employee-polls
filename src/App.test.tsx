// @ts-nocheck
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { fireEvent, render, screen } from '@testing-library/react';

import store from './store';
import { _saveQuestion, _saveQuestionAnswer } from './store/_DATA';
import { Login, NewQuestion, PageNotFound } from './views';
import { Questions } from './components';

jest.spyOn(window, 'alert').mockImplementation(() => {});

const testUser = {
  id: 'tylermcginnis',
  password: 'abc321',
  name: 'Tyler McGinnis',
  avatarURL: '/assets/avatars/tyler.jpg',
  answers: {
    vthrdm985a262al8qx3do: 'optionOne',
    xj352vofupe1dqz9emx13r: 'optionTwo'
  },
  questions: ['loxhs1bqm25b708cmbf3g', 'vthrdm985a262al8qx3do']
};

const testQuestion = {
  id: '8xf0y6ziyjabvozdd253nd',
  author: 'sarahedo',
  timestamp: 1467166872634,
  optionOne: {
    votes: ['sarahedo'],
    text: 'Build our new application with Javascript'
  },
  optionTwo: {
    votes: [],
    text: 'Build our new application with Typescript'
  }
};

describe('_saveQuestion()', () => {
  const newQuestion = {
    optionOneText: 'True',
    optionTwoText: 'False'
  };
  it('returns new question when correct data is passed', async () => {
    const response = await _saveQuestion({ author: testUser.id, ...newQuestion });
    expect(response).toMatchObject({
      id: expect.any(String),
      author: testUser.id,
      optionOne: {
        text: newQuestion.optionOneText,
        votes: []
      },
      optionTwo: {
        text: newQuestion.optionTwoText,
        votes: []
      }
    });
  });

  it('throws error when incorrect data is passed', async () => {
    await expect(_saveQuestion(newQuestion)).rejects.toEqual(
      'Please provide optionOneText, optionTwoText, and author'
    );
  });
});

describe('_saveQuestionAnswer()', () => {
  const newQuestionAnswer = {
    qid: testQuestion.id,
    answer: 'optionOne'
  };
  it('returns users & questions when correct data is passed', async () => {
    const response = await _saveQuestionAnswer({ authedUser: testUser.id, ...newQuestionAnswer });
    expect(response.users[testUser.id].answers[testQuestion.id]).toEqual(newQuestionAnswer.answer);
    expect(response.questions[testQuestion.id][newQuestionAnswer.answer].votes).toContain(
      testUser.id
    );
  });

  it('throws error when incorrect data is passed', async () => {
    await expect(_saveQuestionAnswer(newQuestionAnswer)).rejects.toEqual(
      'Please provide authedUser, qid, and answer'
    );
  });
});

describe('rendered login page', () => {
  it('matches the snapshot', () => {
    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>
    );
    expect(view).toMatchSnapshot();
  });
});

describe('rendered new question page', () => {
  it('matches the snapshot', () => {
    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <NewQuestion />
        </BrowserRouter>
      </Provider>
    );
    expect(view).toMatchSnapshot();
  });
});

describe('rendered 404 page', () => {
  it('matches the snapshot', () => {
    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <PageNotFound />
        </BrowserRouter>
      </Provider>
    );
    expect(view).toMatchSnapshot();
  });
});

describe('rendered Questions component', () => {
  it('matches the snapshot', () => {
    const view = render(
      <Provider store={store}>
        <BrowserRouter>
          <Questions title='New Questions' questions={[testQuestion]} />
        </BrowserRouter>
      </Provider>
    );
    expect(view).toMatchSnapshot();
  });
});

describe('new question page', () => {
  it('gives alert when empty form is submitted', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <NewQuestion />
        </BrowserRouter>
      </Provider>
    );
    const submitButton = screen.getByTestId('submitNewQuestion');
    submitButton.click();
    expect(window.alert).toBeCalledWith('First option cannot be empty!');
  });

  it('gives alert when partially filled form is submitted', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <NewQuestion />
        </BrowserRouter>
      </Provider>
    );
    const optionOneField = screen.getByTestId('fieldOptionOne').childNodes[0].childNodes[0];
    fireEvent.change(optionOneField, { target: { value: 'True' } });
    const submitButton = screen.getByTestId('submitNewQuestion');
    submitButton.click();
    expect(window.alert).toBeCalledWith('Second option cannot be empty!');
  });
});
