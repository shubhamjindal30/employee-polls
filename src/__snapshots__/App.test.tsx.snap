// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendered login page matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-container-1"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthXs"
        >
          <div
            class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-flex-start MuiGrid-justify-content-xs-center"
            >
              <h5
                class="MuiTypography-root makeStyles-heading-3 MuiTypography-h5"
              >
                Sign in
              </h5>
              <div
                class="MuiFormControl-root makeStyles-usernameInput-4 MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink="false"
                  id="username-select-label"
                >
                  User
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                >
                  <div
                    aria-haspopup="listbox"
                    aria-labelledby="username-select-label username-select"
                    class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                    id="username-select"
                    role="button"
                    tabindex="0"
                  >
                    <span>
                      ​
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput"
                    tabindex="-1"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-9"
                    >
                      <span>
                        Username
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiTextField-root makeStyles-passwordInput-5 MuiFormControl-fullWidth"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-disabled Mui-disabled Mui-required Mui-required"
                  data-shrink="false"
                >
                  Password
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled"
                    disabled=""
                    required=""
                    type="password"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-9"
                    >
                      <span>
                        Password
                         *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-loginBtn-6 MuiButton-containedPrimary MuiButton-fullWidth"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Sign in
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-container-1"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXs"
      >
        <div
          class="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-flex-start MuiGrid-justify-content-xs-center"
          >
            <h5
              class="MuiTypography-root makeStyles-heading-3 MuiTypography-h5"
            >
              Sign in
            </h5>
            <div
              class="MuiFormControl-root makeStyles-usernameInput-4 MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
                id="username-select-label"
              >
                User
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <div
                  aria-haspopup="listbox"
                  aria-labelledby="username-select-label username-select"
                  class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                  id="username-select"
                  role="button"
                  tabindex="0"
                >
                  <span>
                    ​
                  </span>
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput"
                  tabindex="-1"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-9"
                  >
                    <span>
                      Username
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root makeStyles-passwordInput-5 MuiFormControl-fullWidth"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-disabled Mui-disabled Mui-required Mui-required"
                data-shrink="false"
              >
                Password
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled Mui-disabled"
                  disabled=""
                  required=""
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-7 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-9"
                  >
                    <span>
                      Password
                       *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-loginBtn-6 MuiButton-containedPrimary MuiButton-fullWidth"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Sign in
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`rendered new question page matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root makeStyles-container-11 MuiContainer-maxWidthSm"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5"
        >
          Would You Rather
        </h5>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
        >
          Create Your Own Poll
        </p>
        <p
          class="MuiTypography-root makeStyles-firstOption-12 MuiTypography-body1"
        >
          First Option
        </p>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-textField-13 MuiFormControl-fullWidth"
          data-testid="fieldOptionOne"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              placeholder="Option One"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-17"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <p
          class="MuiTypography-root makeStyles-secondOption-14 MuiTypography-body1"
        >
          Second Option
        </p>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-textField-13 MuiFormControl-fullWidth"
          data-testid="fieldOptionTwo"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              placeholder="Option Two"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-17"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submitBtn-15 MuiButton-containedPrimary"
          data-testid="submitNewQuestion"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Submit
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root makeStyles-container-11 MuiContainer-maxWidthSm"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5"
      >
        Would You Rather
      </h5>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        Create Your Own Poll
      </p>
      <p
        class="MuiTypography-root makeStyles-firstOption-12 MuiTypography-body1"
      >
        First Option
      </p>
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-textField-13 MuiFormControl-fullWidth"
        data-testid="fieldOptionOne"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            placeholder="Option One"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-17"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <p
        class="MuiTypography-root makeStyles-secondOption-14 MuiTypography-body1"
      >
        Second Option
      </p>
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-textField-13 MuiFormControl-fullWidth"
        data-testid="fieldOptionTwo"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            placeholder="Option Two"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-17"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submitBtn-15 MuiButton-containedPrimary"
        data-testid="submitNewQuestion"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Submit
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
